step/src/main/java/com/google/sps/OAuth2Utilities.java

```diff
@@ -0,0 +1,97 @@
+// Copyright 2019 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package com.google.sps;
+
+import java.io.FileReader;
+import java.io.FileNotFoundException;
+import java.io.IOException;
+import com.google.api.client.googleapis.auth.oauth2.GoogleClientSecrets;
+import com.google.api.client.googleapis.auth.oauth2.GoogleAuthorizationCodeFlow;
+import com.google.api.client.googleapis.auth.oauth2.GoogleAuthorizationCodeFlow.Builder;
+import com.google.api.client.auth.oauth2.Credential;
+import com.google.api.client.auth.oauth2.StoredCredential;
+import com.google.api.client.extensions.appengine.datastore.AppEngineDataStoreFactory;
+import com.google.api.client.http.javanet.NetHttpTransport;
+import com.google.api.client.json.jackson2.JacksonFactory;
+import com.google.api.client.util.store.DataStore;
+
+public class OAuth2Utilities {
+
+  public final static String CREDENTIAL_DATASTORE_ID = "credential_datastore";
+  private final static String CLIENT_SECRETS_JSON_PATH = "./WEB-INF/classes/client_secrets.json";
+  public final static String AUTH_REDIRECT_URI = // TODO find a way to get rid of this constant
+    "https://8080-7dc48ed0-1a8b-4df6-ba73-e55ccd2fb9ed.us-central1.cloudshell.dev/oauth2";
+    // "https://brenda-ding-pod-step-20.ue.r.appspot.com/oauth2";
```

Comment:
Why is this commented out?

Reply:
AUTH_REDIRECT_URI refers the the URL that the user is redirected to when they're done with Google's OAuth consent screen. When we're working on it locally, it needs to be https://8080 but when it's deployed it needs to be https://brenda-ding. I'm still trying to figure out a way that our code can determine the redirect URL on it's own but I put these 2 URLs here so we wouldn't have to retype them while I look for a better solution.

Reply:
Comments of things you are trying to figure out should not be included. This can live in another PR that you are still working on.